File: Typescript/1_Two_Sum.ts
/**
 * @param nums - The array of integers
 * @param target - The target sum
 * @returns The indices of the two numbers that add up to the target
 */
export function twoSum(nums: number[], target: number): number[] {
    const result: { [key: number]: number } = {};
    for (let i = 0; i < nums.length; i++) {
        const temp = target - nums[i];
        if (result.hasOwnProperty(temp))
            return [result[temp], i];
        result[nums[i]] = i;
    }
    return [];
}

File: Python3/1_Two_Sum.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        """
        Finds two numbers in the array that add up to the target.

        Args:
            nums: A list of integers.
            target: The target sum.

        Returns:
            A list of two indices corresponding to the numbers that add up to the target.
        """
        temp = {}
        for index, value in enumerate(nums):
            j = target - value
            if j in temp:
                return [temp[j], index]
            temp[value] = index
